packages/web3torrent/src/App.tsx

```diff
@@ -11,88 +11,59 @@ import File from './pages/file/File';
 import Upload from './pages/upload/Upload';
 import {RoutePath} from './routes';
 import {Web3TorrentContext} from './clients/web3torrent-client';
-import {TorrentClientCapabilities, ClientEvents} from './library/types';
+import {TorrentClientCapabilities} from './library/types';
+import {requiredNetwork} from './constants';
 
 const history = createBrowserHistory();
-class App extends React.Component {
-  state = {
-    currentNetwork: 'ethereum' in window && Number(window.ethereum.networkVersion),
-    requiredNetwork: Number(process.env.REACT_APP_CHAIN_NETWORK_ID),
-    canTorrent: true,
-    showNetworkWarning: false
-  };
 
-  static contextType = Web3TorrentContext;
+const App: React.FC = () => {
+  const Web3Torrent = useContext(Web3TorrentContext);
+  const [currentNetwork, setCurrentNetwork] = useState(
+    window.ethereum ? Number(window.ethereum.networkVersion) : undefined
+  );
 
-  // Adds typing information to this.context
-  context!: React.ContextType<typeof Web3TorrentContext>;
-
-  constructor(props) {
-    super(props);
-    this.updateState = this.updateState.bind(this);
-  }
-
-  updateState() {
-    this.setState({
-      ...this.state,
-      showNetworkWarning: this.context.clientCapability === TorrentClientCapabilities.NOT_CAPABLE,
-      currentNetwork: Number(window.ethereum.networkVersion)
-    });
-  }
-
-  componentDidMount() {
-    this.context.on(ClientEvents.CLIENT_CAPABILITY_TEST, this.updateState);
-    if ('ethereum' in window) {
-      this.setState({...this.state, currentNetwork: Number(window.ethereum.networkVersion)});
-      window.ethereum.on('networkChanged', this.updateState);
+  useEffect(() => {
+    if (window.ethereum) {
+      window.ethereum.on('networkChanged', chainId => {
+        setCurrentNetwork(Number(chainId));
+      });
     }
-  }
+  }, []);
```

Comment:
Is the approach of adding the window.ethereum.networkVersion to the array of dependencies of the useEffect something you tried?

I think that using that would remove the necessity of maintaining a listener

Reply:
Not sure I fully understand unfortunately. Could you come up with a (pseudo)code of what you are thinking about?

Reply:
Something like
```
useEffect(() => {
    setCurrentNetwork(window.ethereum && Number(window.ethereum.networkVersion));
  }, [window.ethereum, window.ethereum.networkVersion]);
```
This way, React is the one checking for changes inside the dependencies, and executing the function specified inside the useEffect, **and** we don't have to mantain the listener.

Though I don't know if it would actually work, I think I remember a case where a prop was changing in the web3torrent library and the change wasn't being picked up by React

Reply:
Thanks for the clarification, and that's a clever thought. But I think the above doesn't quite work. `useEffect` hook runs after every component render. So `window.ethereum` would only be checked after the `App` component renders. And what if `window.ethereum` changes in the background without triggering `App` component re-rendering?